src/System.IO.Abstractions.TestingHelpers/MockFileSystem.cs

```diff
@@ -88,6 +91,19 @@ public MockFileSystem(IDictionary<string, MockFileData> files, string currentDir
         /// <inheritdoc />
         public PathVerifier PathVerifier => pathVerifier;
 
+        /// <summary>
+        /// Replaces the time provider with a mocked instance. This allows to influence the used time in tests.
+        /// <para />
+        /// If not set, the default implementation returns <see cref="DateTime.Now"/>.
+        /// </summary>
+        /// <param name="dateTimeProvider">The function that returns the current <see cref="DateTime"/>.</param>
+        /// <returns></returns>
+        public MockFileSystem MockTime(Func<DateTime> dateTimeProvider)
```

Comment:
This would introduce a new pattern for altering behavior of the mock file system. Could we maybe just add a new optional constructor parameter to pass the provider?

Reply:
If I add it only as a constructor parameter, I can't change it later on, as I used it in the unit tests.
I could instead make it a public writable property, but don't think, that this is fundamentally a better approach...